src/proposals.jl

```diff
@@ -0,0 +1,10 @@
+abstract type AbstractProposal end
+
+struct RecomProposal <: AbstractProposal
```

Comment:
Just want to make sure I'm clear on the data structures here—what does an `AbstractProposal` eventually become once it's accepted... a `Plan`/`Partition`?